python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -1004,6 +1017,14 @@ def get_partition_keys_not_in_subset(
 
         return result
 
+    def get_partition_keys(self, current_time: Optional[datetime] = None) -> Iterable[str]:
+        partition_keys_not_in_subset = set(self.get_partition_keys_not_in_subset())
+        return [
+            key
+            for key in self._partitions_def.get_partition_keys(current_time=current_time)
```

Comment:
Same comment as above about `get_partition_keys`. Could we instead iterate over the windows in the subset and return the partition keys in each window, without iterating over partition keys that are not in any of the windows?